test/processors/JsonSchemaInputProcessor.spec.ts

```diff
@@ -123,4 +124,79 @@ describe('JsonSchemaInputProcessor', function() {
         });
     });
 
-});
\ No newline at end of file
+    describe('reflectSchemaName()', function() {
+        test('should work', async function() {
+            const schema = {
+                properties: {
+                    prop: {
+                        type: "string",
+                    },
+                    allOfCase: {
+                        allOf: [
+                            {
+                                type: "string",
+                            },
+                            {
+                                type: "string",
+                            },
+                        ],
+                    }, 
+                },
+                patternProperties: {
+                    patternProp: {
+                        type: "string",
+                    }
+                },
+                dependencies: {
+                    dep: {
+                        type: "string",
+                    },
+                },
+                definitions: {
+                    def: {
+                        type: "string",
+                    },
+                    oneOfCase: {
+                        oneOf: [
+                            {
+                                type: "string",
+                            },
+                            {
+                                type: "string",
+                            },
+                        ],
+                    }, 
+                },
+                anyOf: [
+                    {
+                        type: "string",
+                    },
+                    {
+                        type: "string",
+                    },
+                ]
+            }
+            const expected = JsonSchemaInputProcessor.reflectSchemaName(schema) as any;
+
+            // properties
+            expect(expected.properties.prop['x-modelgen-inferred-name']).toEqual('prop');
+            expect(expected.properties.allOfCase.allOf[0]['x-modelgen-inferred-name']).toEqual('allOfCase_allOf_0');
+            expect(expected.properties.allOfCase.allOf[1]['x-modelgen-inferred-name']).toEqual('allOfCase_allOf_1');
+
+            // patternProperties
+            expect(expected.patternProperties.patternProp['x-modelgen-inferred-name']).toEqual('patternProp');
```

Comment:
Are you sure you would use regex as a name?

Reply:
Good point!